packages/caravan-psbt/src/psbtv0/utils.ts

```diff
@@ -143,8 +143,9 @@ export const convertLegacyOutput = (output: LegacyOutput): PsbtOutput => {
     value: new BigNumber(output.amountSats).toNumber(),
     bip32Derivation:
       output.bip32Derivation || getBip32Derivation(output.multisig),
-    witnessScript: output.witnessScript || output.multisig?.witness?.output,
-    redeemScript: output.redeemScript || output.multisig?.redeem?.output,
+    witnessScript: output.witnessScript,
+    redeemScript: output.redeemScript,
+    ...psbtMultisigLock(output.multisig),
```

Comment:
We'll need to make sure to update this function for basically the same chnage you did for `psbtInputFormatter`.

Reply:
A bit confused on this—does this suggest we should have the following in `psbtMultisigLock`:
```js
    case P2WSH:
    case P2SH_P2WSH: // need both
      multisigLock.witnessScript = multisigWitnessScript(multisig).output;
      multisigLock.redeemScript = multisigRedeemScript(multisig)?.output;
      break;
```

Noting the optional `redeem` output, since `multisigRedeemScript` returns null for p2wsh

Reply:
Oh yeah, you know what. I think you're right. The description of the function is:

```
/**
 * Return the locking script for the given `Multisig` object in a PSBT consumable format
 */
```
and the only locking script that matters for P2WSH is the witnessScript, so I think you're right. This is fine.

Reply:
Thanks for verifying—understanding now that this is part of the BIP and not just an implementation detail!